lib/active_replicas/rails4/process_local_connection_handler.rb

```diff
@@ -114,6 +122,18 @@ def clear_proxies!
       def current_thread_id
         Thread.current.object_id
       end
+
+      private
+
+      # Sets up `@primary_pool` and `@replica_pools` from the current
+      # `@proxy_configuration`.
+      def initialize_pools
```

Comment:
@dannygreg: I moved it out of the original call site in `#initialize` since it's now used in both `#initialize` and `#establish_connection`. Open to suggestions for a better way to abstract this behavior!